### ISSUE TITLE
[Idea] pull agg out

### ISSUE THREAD

**Issue by mathisloge** [mapnik/mapnik#4339]
@artemp 

I think it would be benefitical to pull agg out of the source code and put it into its own repository. E.g. mapnik/agg. 
We could use the original commit history from agg itself http://agg.sourceforge.net/antigrain.com/index.html and then use the commit history of mapnik and apply it to the original source code. 

It would enable others to use the agg code as a independent code and would make the initial repo size a bit smaller.

**Comment by artemp** [comment]
@mathisloge good idea ^^ . So are you thinking adding agg as a submodule? Lets coordinate how best to accomplish this :+1:

**Comment by mathisloge** [comment]
Either as a submodule or we add a correct installation cmake logic to the agg library. And then it can just be found with `find_package(agg CONFIG required)`.

**Comment by artemp** [comment]
@mathisloge I think we should stick to submodule as `find_package` might find system libagg which is not going to work as expected.  On `Ubuntu 22.04` I get - 

```
sudo apt search libagg
sudo: unable to resolve host brora: Name or service not known
Sorting... Done
Full Text Search... Done
libagg-dev/jammy 1:2.6.1-r134+dfsg1-2 amd64
  AntiGrain Geometry graphical toolkit (static development files)

libagg2/jammy 1:2.6.1-r134+dfsg1-2 amd64
  AntiGrain Geometry graphical toolkit (runtime files)

libagg2-dev/jammy 1:2.6.1-r134+dfsg1-2 amd64
  AntiGrain Geometry graphical toolkit (static and shared development files)
```

**Comment by artemp** [comment]
> probably those won't be found with find_package since they aren't exported with cmake, e.g. there isn't a aggConfig.cmake besides those others. But yeah, we could add it just as a submodule but make it optional like the other submodules, so agg can be provided as a seperate module in package managers :)

Sounds good

**Comment by mathisloge** [comment]
Would you like to create the repo? I'll then move the original source code to it and afterwards apply the modified code from mapnik 👍

**Comment by artemp** [comment]
@mathisloge Sure, I'll do it tomorrow and ping you.

**Comment by artemp** [comment]
https://github.com/mapnik/agg 
@mathisloge should be good to go

**Comment by mathisloge** [comment]
Nice, thanks. 👍 Will do it on sunday
